src/data_conn.c

```diff
@@ -1720,36 +1723,44 @@ uzfs_zvol_rebuild_scanner(void *arg)
 						goto exit;
 					}
 					mutex_enter(&zv->rebuild_mtx);
-					if (zinfo->is_snap_inprogress) {
+					if (zinfo->is_snap_inprogress ||
+					    zinfo->disallow_snapshot) {
+						mutex_exit(&zv->rebuild_mtx);
 						sleep(1);
 						LOG_INFO("waiting for snapshot"
-						    "to finish");
-						mutex_exit(&zv->rebuild_mtx);
+						    " to finish");
 						goto retry;
 					}
-					zinfo->is_afs_inprogress = 1;
+					zinfo->disallow_snapshot = 1;
 					mutex_exit(&zv->rebuild_mtx);
 					uzfs_zvol_send_zio_cmd(zinfo, &hdr,
 					    ZVOL_OPCODE_REBUILD_ALL_SNAP_DONE,
 					    fd, NULL, 0, 0, warg);
 					all_snap_done = B_TRUE;
+					/*
+					 * wait for the downgraded replica to
+					 * transition into the AFS mode. After
+					 * that we can reset the afs_inprogress
```

Comment:
Since we changed `afs_inprogress` to `disallow_snapshot`, one minor correction required here.